### DISCUSSION TITLE
Timezone Client Hints (Server Render dates without the pain!)

### DISCUSSION THREAD

**Discussion by kentcdodds** [epicweb-dev/epic-stack#307] [Announcements]
I'm excited to announce the Epic Stack now has support for a client hint cookie for the user's timezone and a utility to get a date time formatter that accounts for the user's timezone and preferred local!

Check these demos:

https://github.com/epicweb-dev/epic-stack/assets/1500684/21e9c615-65b7-47c9-b448-79d6049cc6e2

https://github.com/epicweb-dev/epic-stack/assets/1500684/7014e19e-858d-474a-8d85-47db0e48288b

That time is server rendered in the user's preferred timezone and locale format. Very cool.

Docs are here: https://github.com/epicweb-dev/epic-stack/blob/main/docs/timezone.md

Thanks to @kiliman and @rajeshg for helping me with this feature!

Here's the commit: https://github.com/epicweb-dev/epic-stack/commit/5e15266d62b0799793c32c2d4ba7ecc09ab49040

**Comment by karlhorky** [comment]
If you would submit an RFC, issue or feature request to the standard, definitely would subscribe! Time zone information would be great to get in the client request :+1: probably also need to consider security / privacy / fingerprinting considerations, but maybe this is general enough to be less of a concern

**Comment by karlhorky** [comment]
Looks like @lucacasonato created a new proposal for HTML which could avoid the need for hints / SSR 👀

> #### Proposed solution
> We propose to extend the `<time>` element with the capability to render a formatted absolute date, time, or date time instead of it's child text node when the user opts into this behaviour using an attribute. We propose that the formatted value is rendered through a UA shadow root.
> 
> ```html
> <time datetime="2026-06-17T08:13:41.099Z" format="datetime">
>   (fallback display)
> </time>
> ```

- https://github.com/whatwg/html/issues/12591
- https://bsky.app/profile/lcas.dev/post/3moi3kdzctc2t
